docs/learn/documentation/versioned/hadoop/overview.md

```diff
@@ -19,28 +19,28 @@ title: Batch Processing Overview
+Samza provides a unified data processing model for both stream and batch processing. The primary difference between batch and streaming is whether the input size is bounded or unbounded. Batch data sources are typically bounded (e.g. static files on HDFS), whereas streams are unbounded (e.g. a topic in Kafka). Under the hood, the same highly-efficient stream-processing engine handles both types.
 
-![diagram-medium](/img/{{site.version}}/learn/documentation/hadoop/unified_batch_streaming.png)
+<img src="/img/{{site.version}}/learn/documentation/hadoop/unified_batch_streaming.png" alt="Unified Batch and Streaming" style="max-width: 100%; height: auto;" onclick="window.open(this.src)">
 
 ### Unified API for Batch and Streaming
 
-Samza provides a single set of APIs for both batch and stream processing. This unified programming API makes it convenient for you to focus on the processing logic, without treating bounded source and unbounded source differently. Switching between batch and streaming only requires config change, e.g. [Kafka](../api/overview.html) to [HDFS](./consumer.html), instead of any code change.
+Samza provides a single set of APIs for both batch and stream processing. This unified programming API makes it convenient for you to focus on the processing logic, without treating bounded and unbounded sources differently. Switching between batch and streaming only requires config change, e.g. [Kafka](../api/overview.html) to [HDFS](./consumer.html), instead of any code change.
 
 ### Multi-stage Batch Pipeline
 
-Complex data pipelines usually consist multiple stages, with data shuffled (repartitioned) across stages for various windowing, aggregation and join. Samza [high-level API](/startup/preview/index.html) provides a useful operator named `partitionBy` to help the repartitioning of the data. Internally, Samza creates a physical stream, called an “intermediate stream”, sends the previous stage output to the stream, and then feed it to the next stage of the pipeline. The lifecycle of intermediate streams is completely managed by Samza so from the user perspective the data shuffling is automatic.
+Complex data pipelines usually consist multiple stages, with data shuffled (repartitioned) between stages to enable key-based operations such as windowing, aggregation, and join. Samza [high-level API](/startup/preview/index.html) provides an operator named `partitionBy` to create such multi-stage pipelines. Internally, Samza creates a physical stream, called an “intermediate stream”, based on the system configured as in `job.default.system`. Samza repartitioned the previous stage output to the intermediate stream, and then feed it to the next stage of the pipeline. The lifecycle of intermediate streams is completely managed by Samza so from the user perspective the data shuffling is automatic.
```

Comment:
s/Samza repartitioned the previous stage output to the intermediate stream, and then feed
/Samza repartitions the output of the previous stage by sending it to the intermediate stream with the appropriate partition count and partition key. It then feeds

Reply:
Fixed. Thanks.